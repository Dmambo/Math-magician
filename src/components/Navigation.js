import React from 'react';
import { NavLink } from 'react-router-dom';
import './styles/Navigation.css';

const Navigation = () => (
  <nav>
    <h1 className="navHead">Math Magicians</h1>
    <div className="list">
      <NavLink exact="true" to="/" className="link active">
        Home
      </NavLink>
      <span className="vr" />
      <NavLink to="/calculator" className="link active">
        Calculator
      </NavLink>
      <span className="vr" />
      <NavLink to="/quote" className="link active">
        Quote
      </NavLink>
    </div>
  </nav>
);

export default Navigation;
