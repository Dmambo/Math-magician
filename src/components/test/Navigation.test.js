import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Navigation from '../Navigation';

test('Navigation renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
