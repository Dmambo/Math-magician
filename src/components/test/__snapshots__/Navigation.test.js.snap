// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<nav>
  <h1
    className="navHead"
  >
    Math Magicians
  </h1>
  <div
    className="list"
  >
    <a
      activeClassName="active"
      aria-current="page"
      className="link active"
      exact={true}
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <span
      className="vr"
    />
    <a
      activeClassName="active"
      className="link"
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <span
      className="vr"
    />
    <a
      activeClassName="active"
      className="link"
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </div>
</nav>
`;
